### ISSUE TITLE
Comma seperated list of training files causes intermittent inf value for batch loss

### ISSUE THREAD

**Issue by David-Levinthal** [mozilla/DeepSpeech#1269]
For support and discussions, please use our [Discourse forums](https://discourse.mozilla.org/c/deep-speech).

If you've found a bug, or have a feature request, then please create an issue with the following information:

- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (our builds, or upstream TensorFlow)**:
- **TensorFlow version (use command below)**:
- **Python version**: 
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can obtain the TensorFlow version with

```bash
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"
```

Please describe the problem clearly. Be sure to convey here why it's a bug or a feature request.

Include any logs or source code that would be helpful to diagnose the problem. For larger logs, link to a Gist, not a screenshot. If including tracebacks, please include the full traceback. Try to provide a reproducible test case.

This occurred on both the r1.4 and r1.5 version. Ubuntu 16.04 cuda 9.1 cudnn 7.0.4 everything built from source (native client, TF)
this fails
python -u DeepSpeech.py --train_files ../Common_voice/cv-valid-train.csv,../Common_voice/cv-other-train.csv --dev_files ../Common_voice/cv-valid-dev.csv --test_files ../Common_voice/cv-valid-test.csv --train_batch_size 1 --dev_batch_size 1 --test_batch_size 1 --n_hidden 2048 --epoch 50 --checkpoint_dir /home/levinth/DS_checkpoint --summary_dir ../DS_summary --summary_secs 10 --learning_rate 0.0001 --dropout_rate 0.2367 --default_stddev 0.046875 > deepspeech1_2048_2.log 2>&1
I added some debug printout (see the Deepspeech discussion about printing periodic progress)
grep inf deepspeech1_2048_2.log
 current_step = 40, batch_loss = inf 
current_step = 100 time_delta = 26.542353 avg batch_loss = inf 
 current_step = 217, batch_loss = inf 
the fix was to use only 1 train file:
python -u DeepSpeech.py --train_files ../Common_voice/cv-valid-train.csv --dev_files ../Common_voice/cv-valid-dev.csv --test_files ../Common_voice/cv-valid-test.csv --train_batch_size 1 --dev_batch_size 1 --test_batch_size 1 --n_hidden 2048 --epoch 50 --checkpoint_dir /home/levinth/DS_checkpoint --summary_dir ../DS_summary --summary_secs 10 --learning_rate 0.0001 --dropout_rate 0.2367 --default_stddev 0.046875 > deepspeech1_2048_4.log 2>&1

**Comment by reuben** [comment]
The use of several training files has nothing to do with the loss diverging. You're changing the training data, which can cause the training process to diverge. You need to tweak your hyperparameters until you find something that works. As a first step I suggest increasing the batch size or lowering the learning rate if your batch size is constrained by the available memory.

And I think other people have had problems with using the "other" split of Common Voice, as it can be way off from the transcriptions. So you might want to stick with the "valid" part, or switch to a speech corpus that's known to work with a set of hyperparameters, like LibriSpeech (see bin/run-librivox.sh).
